diff --git a/node_modules/@react-native-menu/menu/android/src/main/java/com/reactnativemenu/MenuView.kt b/node_modules/@react-native-menu/menu/android/src/main/java/com/reactnativemenu/MenuView.kt
index d30534e..8310694 100644
--- a/node_modules/@react-native-menu/menu/android/src/main/java/com/reactnativemenu/MenuView.kt
+++ b/node_modules/@react-native-menu/menu/android/src/main/java/com/reactnativemenu/MenuView.kt
@@ -55,6 +55,7 @@ class MenuView(private val mContext: ReactContext): ReactViewGroup(mContext) {
   override fun onDetachedFromWindow() {
     super.onDetachedFromWindow()
     if (mIsMenuDisplayed) {
+      mContext.emitDeviceEvent("CONTEXT_MENU_CLOSED")
       mPopupMenu.dismiss()
     }
   }
@@ -230,6 +231,7 @@ class MenuView(private val mContext: ReactContext): ReactViewGroup(mContext) {
         i++
       }
       mPopupMenu.setOnDismissListener {
+        mContext.emitDeviceEvent("CONTEXT_MENU_CLOSED")
         mIsMenuDisplayed = false
       }
       mIsMenuDisplayed = true
