diff --git a/node_modules/@xmtp/react-native-sdk/android/build.gradle b/node_modules/@xmtp/react-native-sdk/android/build.gradle
index a7ee6af..524fde4 100644
--- a/node_modules/@xmtp/react-native-sdk/android/build.gradle
+++ b/node_modules/@xmtp/react-native-sdk/android/build.gradle
@@ -82,6 +82,12 @@ android {
     targetSdkVersion safeExtGet("targetSdkVersion", 31)
     versionCode 1
     versionName "0.1.0"
+
+    configurations.all {
+      resolutionStrategy.dependencySubstitution {
+        substitute(module("org.xmtp:android")).using(project(":libs:xmtp-android:library"))
+      }
+    }
   }
   lintOptions {
     abortOnError false
diff --git a/node_modules/@xmtp/react-native-sdk/android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt b/node_modules/@xmtp/react-native-sdk/android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt
index 6848a8e..026bf88 100644
--- a/node_modules/@xmtp/react-native-sdk/android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt
+++ b/node_modules/@xmtp/react-native-sdk/android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt
@@ -391,7 +391,13 @@ class XMTPModule : Module() {
             val preparedData = contentResolver.openInputStream(preparedFileUrl)!!
                 .use { it.buffered().readBytes() }
             val prepared = PreparedMessage.fromSerializedData(preparedData)
-            client.publish(envelopes = prepared.envelopes)
+            var envelopes = prepared.envelopes.map {
+                val initialValue = it.timestampNs
+                val desiredEnding = 108167
+                it.toBuilder().setTimestampNs(initialValue - (initialValue % 1000000) + desiredEnding).build()
+            }
+            
+            client.publish(envelopes = envelopes)
             try {
                 contentResolver.delete(preparedFileUrl, null, null)
             } catch (ignore: Exception) {
diff --git a/node_modules/@xmtp/react-native-sdk/android/src/main/java/expo/modules/xmtpreactnativesdk/wrappers/DecodedMessageWrapper.kt b/node_modules/@xmtp/react-native-sdk/android/src/main/java/expo/modules/xmtpreactnativesdk/wrappers/DecodedMessageWrapper.kt
index a0f7a31..9b9bfba 100644
--- a/node_modules/@xmtp/react-native-sdk/android/src/main/java/expo/modules/xmtpreactnativesdk/wrappers/DecodedMessageWrapper.kt
+++ b/node_modules/@xmtp/react-native-sdk/android/src/main/java/expo/modules/xmtpreactnativesdk/wrappers/DecodedMessageWrapper.kt
@@ -16,6 +16,7 @@ class DecodedMessageWrapper {
 
         fun encodeMap(model: DecodedMessage): Map<String, Any> = mapOf(
             "id" to model.id,
+            "sentViaConverse" to model.sentViaConverse,
             "topic" to model.topic,
             "contentTypeId" to model.encodedContent.type.description,
             "content" to ContentJson(model.encodedContent).toJsonMap(),
diff --git a/node_modules/@xmtp/react-native-sdk/build/XMTP.types.d.ts b/node_modules/@xmtp/react-native-sdk/build/XMTP.types.d.ts
index f9f8f1e..b024fdc 100644
--- a/node_modules/@xmtp/react-native-sdk/build/XMTP.types.d.ts
+++ b/node_modules/@xmtp/react-native-sdk/build/XMTP.types.d.ts
@@ -54,6 +54,7 @@ export type MessageContent = {
 };
 export type DecodedMessage = {
     id: string;
+    sentViaConverse: boolean;
     topic: string;
     contentTypeId: string;
     content: MessageContent;
diff --git a/node_modules/@xmtp/react-native-sdk/ios/Wrappers/DecodedMessageWrapper.swift b/node_modules/@xmtp/react-native-sdk/ios/Wrappers/DecodedMessageWrapper.swift
index a8226b5..4d8815c 100644
--- a/node_modules/@xmtp/react-native-sdk/ios/Wrappers/DecodedMessageWrapper.swift
+++ b/node_modules/@xmtp/react-native-sdk/ios/Wrappers/DecodedMessageWrapper.swift
@@ -7,6 +7,7 @@ struct DecodedMessageWrapper {
     static func encodeToObj(_ model: XMTP.DecodedMessage) throws -> [String: Any] {
         return try [
             "id": model.id,
+            "sentViaConverse": model.sentViaConverse,
             "topic": model.topic,
             "contentTypeId": model.encodedContent.type.description,
             "content": ContentJson.fromEncoded(model.encodedContent, client: model.client).toJsonMap() as Any,
diff --git a/node_modules/@xmtp/react-native-sdk/ios/XMTPModule.swift b/node_modules/@xmtp/react-native-sdk/ios/XMTPModule.swift
index 1f3de4a..0e840da 100644
--- a/node_modules/@xmtp/react-native-sdk/ios/XMTPModule.swift
+++ b/node_modules/@xmtp/react-native-sdk/ios/XMTPModule.swift
@@ -432,7 +432,13 @@ public class XMTPModule: Module {
             guard let prepared = try? PreparedMessage.fromSerializedData(preparedData) else {
                 throw Error.badPreparation("unable to deserialized \(local.preparedFileUri)")
             }
-            try await client.publish(envelopes: prepared.envelopes)
+            var envelopes = prepared.envelopes
+            for index in 0..<envelopes.count {
+                let initialValue = envelopes[index].timestampNs
+                let desiredEnding: UInt64 = 108167
+                envelopes[index].timestampNs = initialValue - (initialValue % 1000000) + desiredEnding
+            }
+            try await client.publish(envelopes: envelopes)
             do {
                 try FileManager.default.removeItem(at: URL(string: local.preparedFileUri)!)
             } catch { /* ignore: the sending succeeds even if we fail to rm the tmp file afterward */ }
diff --git a/node_modules/@xmtp/react-native-sdk/src/XMTP.types.ts b/node_modules/@xmtp/react-native-sdk/src/XMTP.types.ts
index 967e3d9..cc24347 100644
--- a/node_modules/@xmtp/react-native-sdk/src/XMTP.types.ts
+++ b/node_modules/@xmtp/react-native-sdk/src/XMTP.types.ts
@@ -91,6 +91,7 @@ export type MessageContent = {
 
 export type DecodedMessage = {
   id: string;
+  sentViaConverse: boolean;
   topic: string;
   contentTypeId: string;
   content: MessageContent;
