diff --git a/node_modules/eth-block-tracker/src/polling.js b/node_modules/eth-block-tracker/src/polling.js
index 1cc8ba7..5f1123e 100644
--- a/node_modules/eth-block-tracker/src/polling.js
+++ b/node_modules/eth-block-tracker/src/polling.js
@@ -60,9 +60,12 @@ class PollingBlockTracker extends BaseBlockTracker {
   }
 
   async _updateLatestBlock () {
-    // fetch + set latest block
-    const latestBlock = await this._fetchLatestBlock()
-    this._newPotentialLatest(latestBlock)
+    try {
+      const latestBlock = await this._fetchLatestBlock()
+      this._newPotentialLatest(latestBlock)
+    } catch (e) {
+      console.log(e);
+    }
   }
 
   async _fetchLatestBlock () {
