diff --git a/node_modules/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.m b/node_modules/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.m
index 5d5d308..92a5b40 100644
--- a/node_modules/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.m
+++ b/node_modules/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.m
@@ -303,4 +303,18 @@ - (CGRect)caretRectForPosition:(UITextPosition *)position
 
 #pragma mark - Utility Methods
 
+- (NSArray *)keyCommands {
+    return @[[UIKeyCommand keyCommandWithInput:@"\r" modifierFlags:UIKeyModifierShift action:@selector(shiftEnter:)]];
+}
+
+
+- (void)shiftEnter:(id)sender {
+    NSRange cursorPosition = [self selectedRange];
+    NSMutableString *newText = [[NSMutableString alloc] initWithString:[super text]];
+    [newText replaceCharactersInRange:cursorPosition withString:@"\n"];
+    NSRange newPosition = NSMakeRange(cursorPosition.location + 1, 0);
+    [super setText:newText];
+    [super setSelectedRange:newPosition];
+}
+
 @end
diff --git a/node_modules/react-native/React/CoreModules/RCTAppState.mm b/node_modules/react-native/React/CoreModules/RCTAppState.mm
index 389aa6d..c3e5cc6 100644
--- a/node_modules/react-native/React/CoreModules/RCTAppState.mm
+++ b/node_modules/react-native/React/CoreModules/RCTAppState.mm
@@ -88,6 +88,13 @@ - (void)startObserving
                                                object:nil];
   }
 
+    if (@available(macCatalyst 14.0, *)) {
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleAppStateDidChange:) name:@"NSWindowDidBecomeMainNotification" object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleAppStateDidChange:) name:@"NSWindowDidResignMainNotification" object:nil];
+    }
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleAppStateDidChange:) name:@"NSWindowDidBecomeMainNotification" object:nil];
+
   [[NSNotificationCenter defaultCenter] addObserver:self
                                            selector:@selector(handleMemoryWarning)
                                                name:UIApplicationDidReceiveMemoryWarningNotification
@@ -112,13 +119,23 @@ - (void)handleAppStateDidChange:(NSNotification *)notification
 {
   NSString *newState;
 
-  if ([notification.name isEqualToString:UIApplicationWillResignActiveNotification]) {
-    newState = @"inactive";
-  } else if ([notification.name isEqualToString:UIApplicationWillEnterForegroundNotification]) {
-    newState = @"background";
-  } else {
-    newState = RCTCurrentAppState();
-  }
+    if ([notification.name isEqualToString:UIApplicationWillResignActiveNotification]) {
+        newState = @"inactive";
+    } else if ([notification.name isEqualToString:UIApplicationWillEnterForegroundNotification]) {
+        newState = @"background";
+    } else {
+        newState = RCTCurrentAppState();
+    }
+    
+    if (@available(macCatalyst 14.0, *)) {
+        if ([notification.name isEqualToString:@"NSWindowDidBecomeMainNotification"]) {
+            newState = @"active";
+        } else if ([notification.name isEqualToString:@"NSWindowDidResignMainNotification"]) {
+            newState = @"background";
+        }
+    }
+
+  
 
   if (![newState isEqualToString:_lastKnownState]) {
     _lastKnownState = newState;
