diff --git a/node_modules/react-native/Libraries/Components/View/ReactNativeStyleAttributes.js b/node_modules/react-native/Libraries/Components/View/ReactNativeStyleAttributes.js
index 926c0c7..f61659f 100644
--- a/node_modules/react-native/Libraries/Components/View/ReactNativeStyleAttributes.js
+++ b/node_modules/react-native/Libraries/Components/View/ReactNativeStyleAttributes.js
@@ -91,6 +91,7 @@ const ReactNativeStyleAttributes: {[string]: AnyAttributeType, ...} = {
   /**
    * Transform
    */
+  scaleY: true,
   transform: {process: processTransform},
 
   /**
diff --git a/node_modules/react-native/Libraries/Lists/VirtualizedList.js b/node_modules/react-native/Libraries/Lists/VirtualizedList.js
index e948a85..c62b25a 100644
--- a/node_modules/react-native/Libraries/Lists/VirtualizedList.js
+++ b/node_modules/react-native/Libraries/Lists/VirtualizedList.js
@@ -47,6 +47,7 @@ import {
 } from './VirtualizeUtils';
 import invariant from 'invariant';
 import * as React from 'react';
+import { Platform } from 'react-native';
 
 export type {RenderItemProps, RenderItemType, Separators};
 
@@ -1843,8 +1844,8 @@ export default class VirtualizedList extends StateSafePureComponent<
 }
 
 const styles = StyleSheet.create({
-  verticallyInverted: {
-    transform: [{scaleY: -1}],
+  verticallyInverted: Platform.OS === "android" ? { scaleY: -1 } : {
+     transform: [{scaleY: -1}],
   },
   horizontallyInverted: {
     transform: [{scaleX: -1}],
