diff --git a/node_modules/react-native-mmkv/android/src/main/cpp/MmkvHostObject.cpp b/node_modules/react-native-mmkv/android/src/main/cpp/MmkvHostObject.cpp
index 1ffbdc2..6b4532c 100644
--- a/node_modules/react-native-mmkv/android/src/main/cpp/MmkvHostObject.cpp
+++ b/node_modules/react-native-mmkv/android/src/main/cpp/MmkvHostObject.cpp
@@ -21,7 +21,7 @@ MmkvHostObject::MmkvHostObject(const std::string& instanceId, std::string path,
                       instanceId.c_str(), path.c_str(), hasEncryptionKey);
   std::string* pathPtr = path.size() > 0 ? &path : nullptr;
   std::string* cryptKeyPtr = cryptKey.size() > 0 ? &cryptKey : nullptr;
-  instance = MMKV::mmkvWithID(instanceId, mmkv::DEFAULT_MMAP_SIZE, MMKV_SINGLE_PROCESS, cryptKeyPtr,
+  instance = MMKV::mmkvWithID(instanceId, mmkv::DEFAULT_MMAP_SIZE, MMKV_MULTI_PROCESS, cryptKeyPtr,
                               pathPtr);
 
   if (instance == nullptr) {
