diff --git a/node_modules/react-native-paper/src/components/FAB/FAB.tsx b/node_modules/react-native-paper/src/components/FAB/FAB.tsx
index 7d06774..c1c0d68 100644
--- a/node_modules/react-native-paper/src/components/FAB/FAB.tsx
+++ b/node_modules/react-native-paper/src/components/FAB/FAB.tsx
@@ -247,7 +247,7 @@ const FAB = forwardRef<View, Props>(
       ...font,
     };
 
-    const md3Elevation = isFlatMode || disabled ? 0 : 3;
+    const md3Elevation = isFlatMode || disabled ? 0 : 1;
 
     const newAccessibilityState = { ...accessibilityState, disabled };
 
