diff --git a/node_modules/react-native-mmkv/android/src/main/cpp/MmkvHostObject.cpp b/node_modules/react-native-mmkv/android/src/main/cpp/MmkvHostObject.cpp
index 1ffbdc2..6b4532c 100644
--- a/node_modules/react-native-mmkv/android/src/main/cpp/MmkvHostObject.cpp
+++ b/node_modules/react-native-mmkv/android/src/main/cpp/MmkvHostObject.cpp
@@ -21,7 +21,7 @@ MmkvHostObject::MmkvHostObject(const std::string& instanceId, std::string path,
                       instanceId.c_str(), path.c_str(), hasEncryptionKey);
   std::string* pathPtr = path.size() > 0 ? &path : nullptr;
   std::string* cryptKeyPtr = cryptKey.size() > 0 ? &cryptKey : nullptr;
-  instance = MMKV::mmkvWithID(instanceId, mmkv::DEFAULT_MMAP_SIZE, MMKV_SINGLE_PROCESS, cryptKeyPtr,
+  instance = MMKV::mmkvWithID(instanceId, mmkv::DEFAULT_MMAP_SIZE, MMKV_MULTI_PROCESS, cryptKeyPtr,
                               pathPtr);
 
   if (instance == nullptr) {
diff --git a/node_modules/react-native-mmkv/ios/MmkvHostObject.mm b/node_modules/react-native-mmkv/ios/MmkvHostObject.mm
index 16045fd..49e3801 100644
--- a/node_modules/react-native-mmkv/ios/MmkvHostObject.mm
+++ b/node_modules/react-native-mmkv/ios/MmkvHostObject.mm
@@ -13,7 +13,7 @@
 #import <vector>
 
 MmkvHostObject::MmkvHostObject(NSString* instanceId, NSString* path, NSString* cryptKey) {
-  NSData* cryptData = cryptKey == nil ? nil : [cryptKey dataUsingEncoding:NSUTF8StringEncoding];
+  NSData* cryptData = cryptKey == nil ? nil : [[NSData alloc] initWithBase64EncodedString:cryptKey options:0];;
 
   // Get appGroup value from info.plist using key "AppGroup"
   NSString* appGroup = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"AppGroup"];
